internal/resource/config/localidentity/identityprofile/local_identity_identity_profile_common.go

```diff
@@ -0,0 +1,176 @@
+package localidentity
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/resourcelink"
+)
+
+var (
+	authSourcesAttrTypes = map[string]attr.Type{
+		"id":     basetypes.StringType{},
```

Comment:
This may be better for another ticket, but I think we ought to be using `types.StringType` (and similar for other types) rather than the `basetypes` version. By doing it this way with `basetypes`, we are instantiating a new struct for every one of these attributes. `types.StringType` avoids that and uses the same existing struct for each occurrence.

Reply:
I have added another case to handle this.